### DISCUSSION TITLE
Help around

### DISCUSSION THREAD

**Discussion by eladm-ultrawis** [paullouisageneau/libdatachannel#922] [Q&A] [answered]
I have a c++ application running locally that captures live video frames from a camera.
My end goal is to be able to stream the video frames to a web app (running on a aws server) that will display the live video to clients running a browser.

I am new to everything related to using webrtc, and web development in general. My main issue is that I am not sure which components I need, and where (in which application or endpoint) one should reside.

My initial thinking is:
- I need some kind of "streaming server" that resides on the local C++ application. captures frames and listen to requests from the web application.
- In my web app (aws), do I need to setup some sort of receiver?

I know this is probably a very broad thing to ask, but I am lacking the basic knowledge to ask the right questions. so any directions or advice will be welcomed.

Thanks!

**Comment by paullouisageneau** [comment, marked as answer]
Your thinking is correct.

libdatachannel can be used for the C++ streaming server, and you'll need to implement a simple WebRTC receiver on browser side.

You should look at the media-sender example for a very basic exemple forwarding RTP packets, and at the streamer exemple for a more complex one where libdatachannel reads raw h264 frames and does the RTP packetization internally.

**Reply by eladm-ultrawis** [reply]
Thanks!

If you don't mind me asking, if my streaming source has no "fixed IP" (it cannot be used as a server). can the samples in libdatachannel work in a scenario of "client side streaming"?

**Reply by paullouisageneau** [reply]
Yes, it will work, since WebRTC establishes a peer-to-peer connection.
